crates/ra_ssr/src/resolving.rs

```diff
@@ -157,6 +170,18 @@ impl<'db> ResolutionScope<'db> {
         ResolutionScope {
             scope,
             hygiene: hir::Hygiene::new(sema.db, resolve_context.file_id.into()),
+            node,
+        }
+    }
+
+    /// Returns the function in which SSR was invoked, if any.
+    pub(crate) fn current_function(&self) -> Option<SyntaxNode> {
+        let mut node = self.node.clone();
+        loop {
+            if node.kind() == SyntaxKind::FN {
+                return Some(node);
+            }
+            node = node.parent()?;
```

Comment:
Could this loop use `node.ancestors()` instead?

Reply:
Thanks. I'd assumed that node.ancestors() started with the first parent, so didn't use it, but you're right, it starts with the node itself.